### PR TITLE
feat(TFD-2146): add ability to pass through PTransform name

### PR DESCRIPTION
**Pull request by bchen-talend**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
add ability to pass through PTransform name

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/filterrow/FilterRowRuntime.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public void build(BeamJobContext ctx) {
 				hasFlow = !StringUtils.isEmpty(flowLink);
 				hasReject = !StringUtils.isEmpty(rejectLink);
 
-				PCollectionTuple outputTuples = expand(mainPCollection);
+				PCollectionTuple outputTuples = mainPCollection.apply(ctx.getPTransformName(), this);
```
Comment (lbourgeois) [inline components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/filterrow/FilterRowRuntime.java]: Don't we need to do same changes on other components runtime ? ex https://github.com/Talend/components/blob/master/components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/typeconverter/TypeConverterRuntime.java#L32

Comment (bchen-talend) [reply components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/filterrow/FilterRowRuntime.java]: Yes @lbourgeois good point, I will have a PR after, to add a proper name for all the PTransform.apply on all component runtime, 
as PTransform is hierarchy invoke, this PR with the other two PRs solve the problem about transfer user defined name to backend.
but the child PTransform like TypeConverterRuntime(which has a well defined name already) is not controlled by user.
I will modify others like KafkaInputPTransformRuntime soon. (just want to other relate PRs be reviewd ASAP)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1450/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1450/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Abchen%2FTFD-2146-PTransform-name)

File: components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/filterrow/FilterRowRuntime.java (hunk 1 above)
Comment (lbourgeois) [reply components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/filterrow/FilterRowRuntime.java]: I don't understand 
> the child PTransform like TypeConverterRuntime(which has a well defined name already) is not controlled by user.
- why are you talking about a child PTransform in that case ?
- the name is currently a generic one _TypeConverter_ so does not reflect the graphical label

Comment (lbourgeois) [reply components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/filterrow/FilterRowRuntime.java]: Ok I got the difference, we should implement _ BeamJobBuilder_ for TypeConverterRuntime if we want access to the context and be able to get the user defined label. I wonder if there isn't a more generic way to do it in the upper classes

Comment (bchen-talend) [reply components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/filterrow/FilterRowRuntime.java]: yes exactly @lbourgeois, refer to https://github.com/Talend/data-streams/pull/496

**Comment by bchen-talend** [comment]
Hi @StefanLMNG @lbourgeois 
I have done a modification about unify the chilid PTransform name, 
I intend to use the build-in naming mechanism by Beam rather than add other names, as it readable and easy to locate the relate code, will send you pic on slack
1 use the class name of PTransform, 
2 give a proper name for anonymous PTransform

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1453/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1453/)

**Comment by bchen-talend** [comment]
failed by salesforce, not related
```
com.sforce.soap.partner.fault.LoginFault
	at org.talend.components.salesforce.runtime.SalesforceSessionReuseTestIT.invalidSession(SalesforceSessionReuseTestIT.java:434)
	at 
```

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1454/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1454/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Abchen%2FTFD-2146-PTransform-name)

**Pull request merged**